### DISCUSSION TITLE
How to store systems in a plugin

### DISCUSSION THREAD

**Discussion by badcodecat** [bevyengine/bevy#11774] [Q&A] [answered]
How can I store systems in a plugin, or is it even possible to do so?
Here's a small example to show what I mean:
```rust
struct ExamplePlugin
{
	system: MysteryType
}

impl ExamplePlugin
{
	fn new(system: MysteryType) -> ExamplePlugin
	{
		ExamplePlugin
		{
			system: system
		}
	}
	
}

impl Plugin for ExamplePlugin
{
	fn build(&self, app: &mut App)
	{
		app.add_systems(Update, system);
	}
}
```
What should I put where I have `MysteryType`?

**Comment by hymm** [comment, marked as answer]
You can store it as `NodeConfigs` and then call `system.into_configs()`.

**Reply by badcodecat** [reply]
What would I use for the generic argument?

**Reply by hymm** [reply]
Ah, my bad. You should use `SystemConfigs`, which is an alias for NodeConfigs with the generic filled in for systems.

**Reply by badcodecat** [reply]
When I use SystemConfigs I get the following error:
mismatched types
expected enum `NodeConfigs<std::boxed::Box<(dyn bevy::prelude::System<In = (), Out = ()> + 'static)>>`
found fn item `for<'a, 'b> fn(bevy::prelude::Commands<'a, 'b>) {test_insert_resource}`
Where `test_insert_resource` is the system.

**Reply by hymm** [reply]
did you call into_configs on test_insert_resource?

**Reply by badcodecat** [reply]
Ah, I see adding `into_configs()` fixed the issue. Sorry I didn't realize I was supposed to call it there.
